x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/token_count_type.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+import { UseField, NumericField, fieldFormatters } from '../../../../shared_imports';
+
+import {
+  StoreParameter,
+  IndexParameter,
+  DocValuesParameter,
+  BoostParameter,
+  AnalyzerParameter,
+  NullValueParameter,
+} from '../../field_parameters';
+import { EditFieldSection, EditFieldFormRow, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'analyzer':
+    case 'boost': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'null_value': {
+      return field.null_value !== undefined && field.null_value !== '';
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const TokenCountType = ({ field }: Props) => {
+  return (
+    <>
+      <EditFieldSection
+        title={i18n.translate('xpack.idxMgmt.mappingsEditor.tokenCount.analyzerSectionTitle', {
+          defaultMessage: 'Analyzer',
+        })}
+      >
+        <AnalyzerParameter
+          path="analyzer"
+          label={i18n.translate('xpack.idxMgmt.mappingsEditor.tokenCount.analyzerFieldLabel', {
+            defaultMessage: 'Index analyzer',
+          })}
+          defaultValue={field.source.analyzer}
+          useDefaultOptions={false}
+        />
+      </EditFieldSection>
+      <EditFieldSection>
+        <StoreParameter />
+        <IndexParameter hasIndexOptions={false} />
+        <DocValuesParameter />
+      </EditFieldSection>
+
+      <AdvancedSettingsWrapper>
+        <EditFieldSection>
+          {/* boost */}
+          <BoostParameter defaultToggleValue={getDefaultValueToggle('boost', field.source)} />
+
+          {/* null_value */}
+          <NullValueParameter
```

Comment:
The description is not correct, as it does not accept a "string" value

> Accepts a string value which is substituted for any explicit null values.

Reply:
good catch 👍